controllers/vsl.go

```diff
@@ -17,6 +18,68 @@ func (r *VeleroReconciler) ValidateVolumeSnapshotLocations(log logr.Logger) (boo
 	}
 	// TODO: For each VSL, confirm for each selected provider, we have the
 	// needed config values
+
+	//checking for empty values
+	if len(velero.Spec.VolumeSnapshotLocations) == 0 {
+		return false, errors.New("no VolumeSnapshotLocations configured, ensure a VolumeSnapshotLocation has been configured")
+	}
+
+	for _, vslSpec := range velero.Spec.VolumeSnapshotLocations {
+
+		//AWS
+		if(vslSpec.Provider == "aws"){
+			
+			//validation for AWS
+			//in AWS, region is a required field
+			if(len(vslSpec.Config["region"]) == 0){
+				return false, errors.New("region for AWS VSL is not configured, please ensure a region is configured")
+			}
+
+			//checking the aws plugin, if not present, throw warning message
+			if(!contains(velero.Spec.DefaultVeleroPlugins, "aws")){
+				r.Log.Info("VSL for AWS specified, but AWS plugin not present, might be a misconfiguration")
+			}
+
+			//TODO: Add warn/error messages to Velero CR status field
+		}
+
+		//GCP
+		if(vslSpec.Provider == "gcp"){
+
+			//validation for GCP
+			if(len(vslSpec.Config["region"]) == 0){
+				r.Log.Info("region for GCP VSL is not configured, please check if a region might be needed")
+			}
+
+			//checking the gcp plugin, if not present, throw warning message
+			if(!contains(velero.Spec.DefaultVeleroPlugins, "gcp")){
+				r.Log.Info("VSL for GCP specified, but GCP plugin not present, might be a misconfiguration")
+			}
+
+			//TODO: Add warn/error messages to Velero CR status field
```

Comment:
We will want this in before we merge. The logging information is nice but doesn't actually get back to the user. Please include the status updates here. We should be returning an error instead of logging